### PR TITLE
feat(high-contrast): define high contrast theme files and tokens

### PR DESCRIPTION
**Pull request by liamdebeasi**
Issue number: Internal

---------

This PR introduces `high-contrast-light.{always, class, system}.css` and `high-contrast-dark.{always, class, system}.css` files to add high contrast themes to Ionic applications. It also defines new color tokens that meet AAA WCAG criteria for text.

I also added tests to verify that these colors meet said criteria when used against the correct contrast color.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Reviews:

We already have test criteria that verifies a) the colors chosen meet AAA WCAG criteria and b) that the CSS is being generated properly. It would be good to get manual testing for the `.system.css` stylesheet:

1. Open any test file and make sure the high contrast system file is loaded:
```html
<link href="../../../../../css/themes/high-contrast-dark.system.css" rel="stylesheet" />
```
2. Enable high contrast on your computer. For macOS, this can be enabled in Settings > Accessibility > Display > Increase contrast. For iOS, this can be enabled in Settings > Accessibility > Display & Text size > Increase contrast. For Android, the system level toggle is only available in Android 14 in Settings > Accessibility > Color and Motion > Contrast Level.

### CODE REVIEW

File: core/src/utils/test/playwright/page/utils/set-content.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const setContent = async (page: Page, html: string, testInfo: TestInfo, o
         <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0" />
         <link href="${baseUrl}/css/ionic.bundle.css" rel="stylesheet" />
         <link href="${baseUrl}/scripts/testing/styles.css" rel="stylesheet" />
-        ${theme !== 'light' ? `<link href="${baseUrl}/src/themes/test/${theme}.css" rel="stylesheet" />` : ''}
+        ${theme !== 'light' ? `<link href="${baseUrl}/css/themes/${theme}.always.css" rel="stylesheet" />` : ''}
```
Comment (liamdebeasi) [inline core/src/utils/test/playwright/page/utils/set-content.ts:45]: Our standard contrast dark theme can benefit from this change, so I put up https://github.com/ionic-team/ionic-framework/pull/28795 to change this in `feature-8.0`. Once that is merged in I'll sync this branch with `feature-8.0`.

File: core/src/css/themes/high-contrast-dark.scss [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  }
+}
+
+@mixin high-contrast-dark-md-theme() {
+  & {
+    --ion-background-color: #121212;
+    --ion-background-color-rgb: 18, 18, 18;
+    --ion-text-color: #ffffff;
+    --ion-text-color-rgb: 255, 255, 255;
+    --ion-border-color: #222222;
```
Comment (liamdebeasi) [inline core/src/css/themes/high-contrast-dark.scss:126]: I just added these styles as a placeholder. They are subject to change in FW-5863.

File: core/src/themes/test/colors/theme.e2e.ts [hunk 3]
```diff
@@ -147,3 +152,82 @@ configs({ modes: ['md'], directions: ['ltr'], themes: ['light', 'dark'] }).forEa
...
+            `${styleTestHelpers}
+          <main class="ion-color-${color}">
+            <p class="ion-color">Hello World</p>
+          </main>`,
+            config
+          );
+
+          if (color === 'danger') {
+            await page.pause();
+          }
```
Comment (amandaejohnston) [inline core/src/themes/test/colors/theme.e2e.ts:178]: Was this left in intentionally?

**Review by amandaejohnston — APPROVED** [review]
Tested locally on my Android phone, looks good 👍 Just had one question.

**Review by sean-perkins — APPROVED** [review]
This looks good once the `page.pause` that Amanda commented on is removed.

**Pull request merged**